src/modules/systemlib/param/param.c

```diff
@@ -271,17 +261,32 @@ param_notify_changes(bool is_saved)
 param_t
 param_find_internal(const char *name, bool notification)
 {
-	param_t param;
+	param_t middle;
+	param_t front = 0;
+	param_t last = get_param_info_count() - 1;
 
-	/* perform a linear search of the known parameters */
+	/* perform a binary search of the known parameters */
 
-	for (param = 0; handle_in_range(param); param++) {
-		if (!strcmp(param_info_base[param].name, name)) {
+	while (front <= last) {
+		middle = front + (last - front) / 2;
```

Comment:
This fails to find the very last parameter (VT_WV_YAWR_SCL for px4fmu-v4), because middle will never reach the last value. You can initialize `last` as:
```
	param_t last = get_param_info_count();
```
`middle == last` below should then also never be possible if I see that correct.